### PR TITLE
Spec test coverage

### PR DESCRIPTION
**Pull request by jeffmccune**

### CODE REVIEW

**Comment by adrienthebo** [comment]
@jeffmccune is this ready for a preliminary review or would you like to keep working on it a bit more?

**Comment by jeffmccune** [comment]
> @jeffmccune is this ready for a preliminary review or would you like to keep working on it a bit more?

@adrienthebo Yes, this is ready now.  I rebased it onto the current master and bumped the version of the docker image to 0.5.0 since it changed slightly by using a more recently maintained version of bats-assert.

Here's how I run it:

```bash
make docker_build_bats # to get 0.5.0
make docker_bats_parallel
```

Parallel is 30 to 50% faster than make docker_bats, but only a few seconds in practice.

File: test/spec/init_startup_stdlib_spec.bats [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+@test 'stdlib::init() takes no action with multiple calls.' {
+  stdlib::init
+  run stdlib::init # capture output for assertion
+  assert_success
+  assert_output --regexp 'already initialized, no action taken'
+}
+
+@test 'stdlib::init() initializes global variables' {
```
Comment (adrienthebo) [inline test/spec/init_startup_stdlib_spec.bats:33]: I might be missing something, but is this test duplicated?

File: test/spec/cmd_spec.bats [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# limitations under the License.
+
+load "test_helper/load"
+
+@test 'stdlib::cmd(echo foo) is successful' {
+  run stdlib::cmd echo foo
+  assert_success
+  assert_output 'foo'
+}
+
```
Comment (adrienthebo) [inline test/spec/cmd_spec.bats:22]: I put together a test for a negative assertion for `stdlib::cmd`, you can add if if you think it adds value:

```diff
diff --git i/test/spec/cmd_spec.bats w/test/spec/cmd_spec.bats
index 429cabe..4166474 100644
--- i/test/spec/cmd_spec.bats
+++ w/test/spec/cmd_spec.bats
@@ -20,6 +20,11 @@ load "test_helper/load"
   assert_output 'foo'
 }
 
+@test 'stdlib::cmd(false) is unsuccessful' {
+  run stdlib::cmd false
+  assert_failure
+}
+
 @test 'stdlib::cmd() logs BEGIN, END, and exit_code via debug' {
   DEBUG=1 run stdlib::cmd echo foo
   assert_success
```

**Review by adrienthebo — APPROVED** [review]
This looks really good. I've added one suggestion for an additional test and pointed out a potentially duplicated test, but we can merge this as-is. @jeffmccune make the changes if you think they're worthwhile, and ping me and Morgante when you want this merged.

File: test/spec/cmd_spec.bats (hunk 2 above)
Comment (jeffmccune) [reply test/spec/cmd_spec.bats:22]: Great!  Added in 08d1779...5322b00

File: test/spec/init_startup_stdlib_spec.bats (hunk 1 above)
Comment (jeffmccune) [reply test/spec/init_startup_stdlib_spec.bats:33]: Yes, removed in 08d1779...5322b00

**Comment by jeffmccune** [comment]
This should be ready for merge now.  I added Adrien's feedback and rebased in 5322b00...23e6041

**Review by morgante — APPROVED** [review]
LGTM

**Pull request merged**